drizzlepac/hapsequencer.py

```diff
@@ -812,3 +825,222 @@ def _get_envvar_switch(envvar_name, default=None):
+    match_list = []
+    for wcs_item in wcs_preference:
+        match_list = fnmatch.filter(grism_wcs_set, wcs_item)
+        if match_list:
+            grism_wcsname = match_list[0]
+            log.info("Proposed WCS for use after Grism/Prism examination: {}".format(grism_wcsname))
+            break
+
+    if not grism_wcsname:
+        log.error("None of the preferred WCS names are present in the common set of WCS names for the Grism/Prism images.")
+        log.error("    There is a problem with this visit")
+        sys.exit(1)
+
+        log.info("\nProposed WCS for use after Grism/Prism examination: {}".format(grism_wcsname))
+
+    # If it is now known there is at least one viable Grism/Prism observation, it is desired to make
+    # the active Grism/Prism WCS be the same for all of the Grism/Prism as well as the direct images
+    # in the visit.  The catch can be if any of the direct images do not have an 'a priori' solution
+    # which matches the Grism/Prism images.  In this case, it will be necessary to "fall back" to
+    # the pipeline WCS solution.
+    log.info("\n***** Direct Image Processing *****")
+    direct_wcsname = ''
+    direct_wcs_set = set()
+    skip_direct_list = []
+    exist_direct_set = False
+    # Loop over all the direct images for this detector in the visit
+    for edp in tdp.edp_list:
+        filename = edp.full_filename
+        hdu = fits.open(filename)
+
+        # Make sure the WCS is up-to-date - doing the update for the
+        # direct images here so there is no need to modify a
+        # pre-existing class which could cause an issue
+        drizcorr = hdu[0].header['DRIZCORR']
+        if drizcorr == "OMIT":
+            updatewcs.updatewcs(filename, use_db=False, verbose=False)
+        else:
+            updatewcs.updatewcs(filename, use_db=True, verbose=False)
+
+        # Get all the WCS names which are common to all of the direct images
+        dict_names = wcsutil.altwcs.wcsnames(filename, ext=1)
+        log.info("WCS solutions for file {} are {}.".format(filename, dict_names))
+        hdu.close()
+        if dict_names:
+            # Initialize a set with wcsnames
+            if not exist_direct_set:
+                direct_wcs_set = set(dict_names.values())
+                exist_direct_set = True
+            # Generate the intersection with the set of existing wcsnames and the wcsnames
+            # from the current image
+            else:
+                direct_wcs_set &= set(dict_names.values())
+
+            # Oops...no common wcsnames
+            if not direct_wcs_set:
+                log.error("There are no common WCS solutions with this direct image {} and previously processed images".format(filename))
+                log.error("    There is a problem with this direct image/visit.")
+                sys.exit(1)
+        # If there are no WCS solutions, the image could be bad (e.g., EXPTIME=0 or EXPFLAG="TDF-DOWN...")
```

Comment:
...or the input image was a _raw.fits file.  This is the most unlikely prospect, but never underestimate users. ;-)

Reply:
Ack.  I did not consider this.  I added a message regarding use of *_raw.fits files.